google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -534,36 +534,42 @@ public boolean delete(String bucket, String blob, BlobSourceOption... options) {
 
   @Override
   public boolean delete(BlobId blob, BlobSourceOption... options) {
-    Opts<ObjectSourceOpt> opts = Opts.unwrap(options).resolveFrom(blob).prepend(defaultOpts);
-    GrpcCallContext grpcCallContext =
-        opts.grpcMetadataMapper().apply(GrpcCallContext.createDefault());
-    DeleteObjectRequest.Builder builder =
-        DeleteObjectRequest.newBuilder()
-            .setBucket(bucketNameCodec.encode(blob.getBucket()))
-            .setObject(blob.getName());
-    ifNonNull(blob.getGeneration(), builder::setGeneration);
-    DeleteObjectRequest req = opts.deleteObjectsRequest().apply(builder).build();
-    GrpcCallContext merge = Utils.merge(grpcCallContext, Retrying.newCallContext());
-    return Boolean.TRUE.equals(
-        Retrying.run(
-            getOptions(),
-            retryAlgorithmManager.getFor(req),
-            () -> {
-              try {
-                storageClient.deleteObjectCallable().call(req, merge);
-                return true;
-              } catch (NotFoundException e) {
-                return false;
-              }
-            },
-            Decoder.identity()));
+    Opts<ObjectSourceOpt> opts = Opts.unwrap(options);
+    try {
+      internalObjectDelete(blob, opts);
+      return true;
+    } catch (NotFoundException e) {
+      return false;
+    }
   }
 
   @Override
   public boolean delete(BlobId blob) {
     return delete(blob, new BlobSourceOption[0]);
   }
 
+  @Override
+  public Void internalObjectDelete(BlobId id, Opts<ObjectSourceOpt> opts) {
```

Comment:
why is this public instead of private? 

Not sure if this should be this in PR given it's mainly docs updates.

Reply:
It is a docs only update, because this was branch off another branch the commit had a distinct sha in this PR after merging the earlier PR to main.

I've rebased, and there is only javadoc update again.